DM-26696: Fixes to fix_header usage

timj opened the pull request:
* Trap bad corrections yaml files
* Add filename and obsid to MetadataTranslator.fix_header class method

timj committed:
Trap bad corrections files

If a YAML corrections file exists but is not in the right format
because it uses lists or only has a comment but no mapping,
ignore it and warn.

timj committed:
Fix translate_header tests to deal with the bad corrections file

timj committed:
Add obsid, instrument and filename to translator fix_header method

This lets people use the information in log messages.

timj committed:
Add line to documentation to point to location of corrections file format

timj committed:
Forward filename through tests to ObservationInfo

To give the option of including it in log messages.

timj committed:
Pass filename through to fix_header call in translate_header

RobertLuptonTheGood commented on python/astro_metadata_translator/tests.py:
```diff
@@ -144,15 +144,19 @@ def assertObservationInfoFromYaml(self, file, dir=None, check_wcs=True,  # noqa:
+                                   wcs_params=wcs_params, **kwargs)
 
-    def assertObservationInfo(self, header, check_wcs=True, wcs_params=None, **kwargs):  # noqa: N802
+    def assertObservationInfo(self, header, filename=None, check_wcs=True,  # noqa: N802
```
I'm being dense, but what's the N802 infraction here?

timj replied:
It's a naming complaint. This package tries to use underscores but in this particular method it's matching the non-standard convention of unittest.TestCase.

RobertLuptonTheGood reviewed (approved):
OK

timj committed:
Add standardized methods for constructing log prefix string

This allows all translators to use the filename+obsid or obsid
depending on the availability of the filename.

timj committed:
Update exception messages to use standardized prefix
